### DISCUSSION TITLE
2024-09-25T09:06:05Z	FATAL	Fatal error

### DISCUSSION THREAD

**Discussion by nimblenitin** [aquasecurity/trivy#7670] [Q&A]
### Description

Getting below error in azure pipeline trivy task. This happens intermittently.

22126857a555: Verifying Checksum
22126857a555: Download complete
1e4db3f86ba5: Verifying Checksum
1e4db3f86ba5: Download complete
22126857a555: Pull complete
1e4db3f86ba5: Pull complete
f5a907a27a4a: Pull complete
Digest: sha256:addfb8fd6b9e520c25b22c61d8aa5d58ecd7879177aa959f952bf4734f4e3f60
Status: Downloaded newer image for aquasec/trivy:latest
2024-09-25T09:06:04Z	DEBUG	No plugins loaded
2024-09-25T09:06:04Z	DEBUG	Default config file "file_path=trivy.yaml" not found, using built in values
2024-09-25T09:06:04Z	DEBUG	Cache dir	dir="/root/.cache/trivy"
2024-09-25T09:06:04Z	DEBUG	Cache dir	dir="/root/.cache/trivy"
2024-09-25T09:06:04Z	DEBUG	Parsed severities	severities=[UNKNOWN LOW MEDIUM HIGH CRITICAL]
2024-09-25T09:06:04Z	DEBUG	Ignore statuses	statuses=[]
2024-09-25T09:06:04Z	DEBUG	There is no valid metadata file	err="unable to open a file: open /root/.cache/trivy/db/metadata.json: no such file or directory"
2024-09-25T09:06:04Z	INFO	[db] Need to update DB
2024-09-25T09:06:04Z	INFO	[db] Downloading DB...	repository="ghcr.io/aquasecurity/trivy-db:2"
2024-09-25T09:06:04Z	DEBUG	No metadata file
2024-09-25T09:06:05Z	FATAL	Fatal error
  - init error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.Run
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:366
  - DB error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.NewRunner
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:118
  - failed to download vulnerability DB:
    github.com/aquasecurity/trivy/pkg/commands/operation.DownloadDB
        /home/runner/work/trivy/trivy/pkg/commands/operation/operation.go:41
  - database download error:
    github.com/aquasecurity/trivy/pkg/db.(*Client).Download
        /home/runner/work/trivy/trivy/pkg/db/db.go:162
  - oci download error:
    github.com/aquasecurity/trivy/pkg/oci.(*Artifact).Download
        /home/runner/work/trivy/trivy/pkg/oci/artifact.go:144
  - failed to fetch the layer:
    github.com/aquasecurity/trivy/pkg/oci.(*Artifact).download
        /home/runner/work/trivy/trivy/pkg/oci/artifact.go:158
  - GET https://ghcr.io/v2/aquasecurity/trivy-db/blobs/sha256:e6ce12d272cdc713b01cdc928bf6ee51f50d488a4252ea84393d5ef67426e1a9: TOOMANYREQUESTS: retry-after: 251.77µs, allowed: 44000/minute
##[error]Failed: Trivy detected problems.
Publishing JSON results...
##[error]Unable to process command '##vso[task.addattachment type=JSON_RESULT;name=trivy0.536304797538341.json;]/tmp/trivy-results-0.7104559655238769.json' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Value cannot be null. (Parameter 'Cannot upload task attachment file, attachment file location is not specified or attachment file does not exist on disk.')
Done!
Finishing: Trivy scan - DEV/SIT 

### Desired Behavior

On each azure pipeline run it should scan sucessfully

### Actual Behavior

Pipeline fails now and then

### Reproduction Steps

```bash
1.Try running the below azure pipeline task for trivy
2.
3.
...
jobs: 
  - job: Trivy
    displayName: Trivy
    pool:
      vmImage: 'ubuntu-latest'
    steps:
    - checkout: self

    # - task: npmAuthenticate@0
    #   displayName: NPM Authentication
    #   inputs:
    #     workingFile: '${{ parameters.npmAuthFile }}'
    #   condition: and(succeeded(), eq('${{ parameters.npmAuthenticate }}', true))
      
    # - task: NuGetAuthenticate@0
    #   displayName: NuGet Authentication
    #   condition: and(succeeded(), eq('${{ parameters.nuGetAuthenticate }}', true))

    - task: trivy@1
      continueOnError: true  
      displayName: Trivy scan - DEV/SIT
      inputs:
        path: .
        debug: true
        exitCode: '0'
        options: '--timeout 10m0s'
      condition: and(succeeded(), or(eq(variables['Build.SourceBranchName'], 'DEV'), eq(variables['Build.SourceBranchName'], 'SIT')))

    - task: trivy@1
      displayName: Trivy scan - UAT/PROD
      inputs:
        path: .
        debug: true
        severities: 'HIGH,CRITICAL,MEDIUM'
        exitCode: '0'
        options: '--timeout 10m0s'
      condition: and(succeeded(), or(eq(variables['Build.SourceBranchName'], 'UAT'), eq(variables['Build.SourceBranchName'], 'PROD')))
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
22126857a555: Verifying Checksum
22126857a555: Download complete
1e4db3f86ba5: Verifying Checksum
1e4db3f86ba5: Download complete
22126857a555: Pull complete
1e4db3f86ba5: Pull complete
f5a907a27a4a: Pull complete
Digest: sha256:addfb8fd6b9e520c25b22c61d8aa5d58ecd7879177aa959f952bf4734f4e3f60
Status: Downloaded newer image for aquasec/trivy:latest
2024-09-25T09:06:04Z	DEBUG	No plugins loaded
2024-09-25T09:06:04Z	DEBUG	Default config file "file_path=trivy.yaml" not found, using built in values
2024-09-25T09:06:04Z	DEBUG	Cache dir	dir="/root/.cache/trivy"
2024-09-25T09:06:04Z	DEBUG	Cache dir	dir="/root/.cache/trivy"
2024-09-25T09:06:04Z	DEBUG	Parsed severities	severities=[UNKNOWN LOW MEDIUM HIGH CRITICAL]
2024-09-25T09:06:04Z	DEBUG	Ignore statuses	statuses=[]
2024-09-25T09:06:04Z	DEBUG	There is no valid metadata file	err="unable to open a file: open /root/.cache/trivy/db/metadata.json: no such file or directory"
2024-09-25T09:06:04Z	INFO	[db] Need to update DB
2024-09-25T09:06:04Z	INFO	[db] Downloading DB...	repository="ghcr.io/aquasecurity/trivy-db:2"
2024-09-25T09:06:04Z	DEBUG	No metadata file
2024-09-25T09:06:05Z	FATAL	Fatal error
  - init error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.Run
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:366
  - DB error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.NewRunner
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:118
  - failed to download vulnerability DB:
    github.com/aquasecurity/trivy/pkg/commands/operation.DownloadDB
        /home/runner/work/trivy/trivy/pkg/commands/operation/operation.go:41
  - database download error:
    github.com/aquasecurity/trivy/pkg/db.(*Client).Download
        /home/runner/work/trivy/trivy/pkg/db/db.go:162
  - oci download error:
    github.com/aquasecurity/trivy/pkg/oci.(*Artifact).Download
        /home/runner/work/trivy/trivy/pkg/oci/artifact.go:144
  - failed to fetch the layer:
    github.com/aquasecurity/trivy/pkg/oci.(*Artifact).download
        /home/runner/work/trivy/trivy/pkg/oci/artifact.go:158
  - GET https://ghcr.io/v2/aquasecurity/trivy-db/blobs/sha256:e6ce12d272cdc713b01cdc928bf6ee51f50d488a4252ea84393d5ef67426e1a9: TOOMANYREQUESTS: retry-after: 251.77µs, allowed: 44000/minute
##[error]Failed: Trivy detected problems.
Publishing JSON results...
##[error]Unable to process command '##vso[task.addattachment type=JSON_RESULT;name=trivy0.536304797538341.json;]/tmp/trivy-results-0.7104559655238769.json' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Value cannot be null. (Parameter 'Cannot upload task attachment file, attachment file location is not specified or attachment file does not exist on disk.')
Done!
Finishing: Trivy scan - DEV/SIT
```

### Operating System

Ubuntu-latest

### Version

```bash
1.5.1
```

### Checklist

- [ ] Run `trivy clean --all`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Duplicate of https://github.com/aquasecurity/trivy/discussions/7538
